_test/types/DecimalTest.php

```diff
@@ -149,6 +149,18 @@ public function valueProvider()
             ['1e15', '1000' . "\xE2\x80\xAF" . 'T', '-1', ',', ' ', true, '', '', true],
             ['1e-21', '0.001' . "\xE2\x80\xAF" . 'a', '-1', ',', ' ', true, '', '', true],
 
+            //format string
+            ['5000', '5 000', '-1', '.', ' ', true, '', '', false, ''],
+            ['5000', '5 000.00', '2', '.', ' ', false, '', '', false, '%s'],
+            ['1.7', '1.70', '-1', '.', ' ', true, '', '', false, '%01.2f'],
+            ['1.7', '1.70' , '-1', '.', ' ', true, '', '', false, '%01.2F'],
+            ['1.7', '0001' , '-1', '.', ' ', true, '', '', false, "%'.04d"],
+            ['15', '1111' , '-1', '.', ' ', true, '', '', false, '%04b'],
+            ['362525200', '3.625e+8' , '-1', '.', ' ', true, '', '', false, '%.3e'],
+            ['362525200', '3.625E+8' , '-1', '.', ' ', true, '', '', false, '%.3E'],
+            ['1', '1' , '-1', '.', ' ', true, '', '', false, '%u'],
+            ['-1', '18446744073709551615' , '-1', '.', ' ', true, '', '', false, '%u'],
```

Comment:
can you also add tests for where the format string is wrong and thus ignored?

Reply:
You are right. The `-` should be in the regex as a character and not as a range. By extending the test I realized that. Thank you!

Reply:
This last test looks interesting. I think it is dependent on the platform and would fail on a 32bit system

Reply:
Of course you're right again!
32bit:
```
php > printf('%u','-1');
4294967295
```
I will remove the last line of the test and adjust the penultimate one. It should only be checked whether `%u` is recognized.